### ISSUE TITLE
AMT scanning network feature request

### ISSUE THREAD

**Issue by cahsdhelpdesk** [Ylianst/MeshCentral#1505]
Good morning.

When you scan the network for AMT devices / machines, the default IP scheme is 192.168.1.0/23 and one has to manually enter the subnet / class they wish to scan each time.

Would it be possible to, as part of the configuration to define multiple / different subnets / classes to scan and then be able to choose from them using a dropdown box?

For example
I configure a list multiple subnets / classes
192.168.15.000/23
192.168.8.000/24
192.168.48.000/16
10.58.12.0/20
172.16.1.0/27

So, when you select the Scan Network.

![image](https://user-images.githubusercontent.com/60752638/85030526-2a5ec300-b143-11ea-92c0-b326ad37f9a9.png)

In the IP Range dialog box, a down arrow is presented and when that is selected the configured list of subnets (list from above) is displayed and the appropriate range can be selected.

![image](https://user-images.githubusercontent.com/60752638/85030696-6560f680-b143-11ea-8d4f-727853eac519.png)

Thanks for your consideration.

**Comment by Ylianst** [comment]
Done. Update to MeshCentral v0.5.68 and add the following in the domain section of the config.json:

```
"AmtScanOptions": [
  "192.168.15.0/23",
  "192.168.8.0/24",
  "192.168.48.0/16",
  "10.58.12.0/20",
  "172.16.1.0/27"
],
```

Restart the server and when you click on the scan box, you should see your options.

![image](https://user-images.githubusercontent.com/1319013/85165075-1e920000-b21a-11ea-9791-1ce6bb434968.png)

Let me know if that works.

**Comment by cahsdhelpdesk** [comment]
Eureka, I can report that it works in Chrome.

![image](https://user-images.githubusercontent.com/60752638/85172272-4f802e00-b236-11ea-8248-9c7eaf513c57.png)

However in FireFox I can see the list and select the subnet I want to scan, however there is no down arrow (as you see in Chrome) in the dialog box and the scan button does not become active to click.

![image](https://user-images.githubusercontent.com/60752638/85172357-848c8080-b236-11ea-9720-c6cb38c135de.png)

Not sure why that is.

Also, iIf I may make another request (as we have many subnets to manage), would it be possible to add a "Friendly or Familiar name " to each subnet?

**Comment by Ylianst** [comment]
Fixed the FireFox button not turning active and added code to ignore everything before the last space, you so can type:

`My Home Network 192.168.50.0/24`

The first part will be ignored. So you can setup your options the way you want them. This will be in MeshCentral v0.5.69 when it's next released.

**Comment by cahsdhelpdesk** [comment]
Everything is working! Thank you so much!

**Comment by cahsdhelpdesk** [comment]
Good morning, I have an update to report.
My first test was to be able to scan and since, at that time, I did not need to add any devices.
However, today I was able to scan but when I select the device and try to add, it will not add.
I have tried this in FF and Chrome.
Any thoughts?

**Comment by Ylianst** [comment]
You are right, fixing it now.

**Comment by Ylianst** [comment]
Just published MeshCentral v0.5.74 with a fix for this. Close this issue if the problem is solved. Thanks.

**Comment by cahsdhelpdesk** [comment]
I tested and I was able to add a device after scanning. Thanks so much!
